### DISCUSSION TITLE
API access for reading/changing/creating feature flags/experiments

### DISCUSSION THREAD

**Discussion by IgnusG** [featurehub-io/featurehub#320] [Q&A] [answered]
Does Featurehub plan to support API access to change feature flags/experiments through authenticated calls?
Let's say you have some sort of secure front-end built with a toggle to turn off all experiments (as an example). I cannot find any information about APIs in the docs nor mentioned in the issues. I found the concept of service accounts with the CHANGE_VALUE permission but I'm unsure how/or if they can be used to achieve this.

**Describe the solution you'd like**
An authorized API to read/change/create features, experiments.

**Describe alternatives you've considered**
Using the service accounts although from the docs I couldn't figure out how this would work.

**Comment by IrinaSouth** [comment, marked as answer]
Hi @IgnusG.  Great question. We do have an API for all of this, it is what the FeatureHub Admin Console is based on. It has a full OpenAPI set of documentation. https://github.com/featurehub-io/featurehub/blob/master/backend/mr-api/mr-api.yaml
Search for _-FeatureService_ to see the API for features for example.

We are not planning to release an SDK for this in the near future, but it could certainly be done.  

Thanks!
